alf/environments/suite_dmlab.py

```diff
@@ -0,0 +1,195 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import gym
+import gin.tf
+from tf_agents.environments import suite_gym
+from tf_agents.environments import wrappers
+from alf.environments.suite_socialbot import ProcessPyEnvironment
+
+# `DeepmindLab` are required,
+#   see `https://github.com/deepmind/lab` to build `DeepmindLab`
+
+try:
+    import deepmind_lab
+except ImportError:
+    deepmind_lab = None
+
+
+def is_available():
+    return deepmind_lab is not None
+
+
+@gin.configurable
+def action_discretize(action_spec,
+                      look_left_right_pixels_per_frame=(-20, 20),
+                      look_down_up_pixels_per_frame=(-10, 10),
+                      strafe_left_right=(-1, 1),
+                      move_back_forward=(-1, 1),
+                      fire=(),
+                      jump=(1, ),
+                      crouch=(1, ),
+                      **kwargs):
+    """Discretize action from action_spec
+
+    TODO: action combinations
+
+    Args:
+        action_spec (list(dict)): action spec
+        look_left_right_pixels_per_frame (iterable|str): look left or look right pixels
+        look_down_up_pixels_per_frame (iterable|str): look down or look up pixels
+        strafe_left_right (iterable|str): strafe left or strafe right
+        move_back_forward (iterable|str): move back or move forward
+        fire (iterable|str): fire values
+        jump (iterable|str): jump values
+        crouch (iterable|str): crouch values
+        kwargs (dict): other config for actions
+    Returns:
+        actions (list[numpy.array]): discrete actions
+    """
+    actions = []
+
+    config = dict(
+        look_left_right_pixels_per_frame=look_left_right_pixels_per_frame,
+        look_down_up_pixels_per_frame=look_down_up_pixels_per_frame,
+        strafe_left_right=strafe_left_right,
+        move_back_forward=move_back_forward,
+        fire=fire,
+        jump=jump,
+        crouch=crouch)
+    config.update(kwargs)
+    config = {key.upper(): value for key, value in config.items()}
+
+    for i, spec in enumerate(action_spec):
+        val_min = spec['min']
+        val_max = spec['max']
+        values = config.get(spec['name'], None)
+
+        if values is None:
+            values = list(range(val_min, val_max + 1))
+        elif isinstance(values, str):
+            values = eval(values)
+
+        for value in values:
+            if value < val_min or value > val_max or value == 0:
```

Comment:
why ignoring value == 0?

Reply:
valid value for action is   `[min, max]  - {0}`, 0 means idle in this direction